src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Extensions/AssertExtensionsTests.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Sdk;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.CollectionUtils;
+using static GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions.AssertExtensions;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+
+public class AssertExtensionsTests
+{
+    private record Company(string Name, List<Person> Employees);
+
+    private record Person(string Name, int Age);
+
+    [Fact]
+    public void AssertDeepEqualsTo_Simple()
+    {
+        var person1 = new Person("Name", 30);
+        var person2 = new Person("Name", 30);
+        person1.AssertDeepEqualTo(person2);
+    }
+
+    [Fact]
+    public void AssertDeepEqualsTo_Simple_NotEqual()
+    {
+        var person1 = new Person("Name", 30);
+        var person2 = new Person("Different", 30);
+        Assert.Throws<TrueException>(() => person1.AssertDeepEqualTo(person2));
+    }
+
+    [Fact]
+    public void AssertDeepEqualsTo_Nested()
+    {
+        var company1 = new Company("Company", ListOf(
+            new Person("Name1", 30),
+            new Person("Name2", 31)));
+
+        var company2 = new Company("Company", ListOf(
+            new Person("Name1", 30),
+            new Person("Name2", 31)));
+
+        company1.AssertDeepEqualTo(company2);
+    }
+
+    [Fact]
+    public void AssertDeepEqualsTo_Nested_NotEqual()
+    {
+        var company1 = new Company("Company", ListOf(
+            new Person("Name1", 30),
+            new Person("Name2", 31)));
+
+        var company2 = new Company("Company", ListOf(
+            new Person("Name1", 30),
+            new Person("DifferentName", 31)));
+
+        Assert.Throws<TrueException>(() => company1.AssertDeepEqualTo(company2));
+    }
+
+    [Fact]
+    public void AssertUtcNow_DateTimesWithinTolerance()
+    {
+        DateTime.UtcNow.AssertUtcNow();
+        DateTime.UtcNow.AddMilliseconds(-1000).AssertUtcNow();
+    }
+
+    [Fact]
+    public void AssertUtcNow_DateTimesOutsideTolerance()
+    {
+        Assert.Throws<InRangeException>(() => DateTime.UtcNow.AddDays(-1501).AssertUtcNow());
+        Assert.Throws<InRangeException>(() => DateTime.UtcNow.AddMilliseconds(-500).AssertUtcNow(withinMillis: 100));
+    }
+
+    [Fact]
+    public void AssertDeepEqualsTo_IgnoreField()
```

Comment:
Shouldn't this be grouped with the other `AssertDeepEqualsTo` tests?

Reply:
Done ta